src/lib/snack-bar/snack-bar.ts

```diff
@@ -140,11 +140,29 @@ export class MdSnackBar {
   /**
    * Creates a new overlay and places it in the correct location.
    */
-  private _createOverlay(): OverlayRef {
+  private _createOverlay(config: MdSnackBarConfig): OverlayRef {
     let state = new OverlayState();
-    state.positionStrategy = this._overlay.position().global()
-        .centerHorizontally()
-        .bottom('0');
+    let positionStrategy = this._overlay.position().global();
+    switch (config.position.horizontal) {
+      case 'left':
+        positionStrategy.left('0');
+        break;
+      case 'right':
+        positionStrategy.right('0');
+        break;
+      case 'center':
+        positionStrategy.centerHorizontally();
+        break;
+    }
+    switch (config.position.vertical) {
+      case 'top':
+        positionStrategy.top('0');
+        break;
+      case 'bottom':
+        positionStrategy.bottom('0');
+        break;
+    }
```

Comment:
Need unit test for using the appropriate position based on the config and ltr/rtl state